src/System.Data.SqlClient/src/System/Data/SqlClient/SqlFileStream.Windows.cs

```diff
@@ -0,0 +1,979 @@
+            // potential exceptions during Close/Finalization. Since our internal buffer is
+            // only a single byte in length, the provided user data will always be cached.
+            // As a result, we need to be sure to flush the data to disk ourselves.
+
+            // calling flush here will mimic the internal control flow of System.IO.FileStream
+            _m_fs.Flush();
+        }
+
+        #endregion
+
+        static private readonly char[] s_invalidPathChars = Path.GetInvalidPathChars();
+
+        // path length limitations:
+        // 1. path length storage (in bytes) in UNICODE_STRING is limited to UInt16.MaxValue bytes = Int16.MaxValue chars
+        // 2. GetFullPathName API of kernel32 does not accept paths with length (in chars) greater than 32766
+        //  (32766 is actually Int16.MaxValue - 1, while (-1) is for NULL termination)
+        // We must check for the lowest value between the the two
+        private const int MaxWin32PathLength = short.MaxValue - 1;
+
+        [Conditional("DEBUG")]
+        static private void AssertPathFormat(string path)
+        {
+            Debug.Assert(path != null);
+            Debug.Assert(path == path.Trim());
+            Debug.Assert(path.Length > 0);
+            Debug.Assert(path.Length <= MaxWin32PathLength);
+            Debug.Assert(path.IndexOfAny(s_invalidPathChars) < 0);
+            Debug.Assert(path.StartsWith(@"\\", StringComparison.OrdinalIgnoreCase));
+            Debug.Assert(!path.StartsWith(@"\\.\", StringComparison.Ordinal));
+        }
+
+        // SQLBUVSTS01 bugs 192677 and 193221: we cannot use System.IO.Path.GetFullPath for two reasons:
+        // * it requires PathDiscovery permissions, which is unnecessary for SqlFileStream since we are dealing with network path
+        // * it is limited to 260 length while in our case file path can be much longer
+        // To overcome the above limitations we decided to use GetFullPathName function from kernel32.dll
+        [ResourceExposure(ResourceScope.Machine)]
+        [ResourceConsumption(ResourceScope.Machine)]
+        static private string GetFullPathInternal(string path)
+        {
+            //-----------------------------------------------------------------
+            // precondition validation should be validated by callers of this method
+            // NOTE: if this method moves elsewhere, this assert should become an actual runtime check
+            // as the implicit assumptions here cannot be relied upon in an inter-class context
+            Debug.Assert(path != null);
+
+            // remove leading and trailing whitespace
+            path = path.Trim();
+            if (path.Length == 0)
+            {
+                throw ADP.Argument(SR.GetString(SR.SqlFileStream_InvalidPath), "path");
+            }
+
+            // check for the path length before we normalize it with GetFullPathName
+            if (path.Length > MaxWin32PathLength)
```

Comment:
As mentioned above, I wouldn't bother. Let GetFullPath give you an error.

Reply:
Code removed.